saltapi/web/api/institutions.py

```diff
@@ -18,3 +21,29 @@ def get_institutions() -> List[Dict[str, Any]]:
     with UnitOfWork() as unit_of_work:
         institution_service = services.institution_service(unit_of_work.connection)
         return institution_service.get_institutions()
+
+
+@router.post(
+    "/",
+    summary="Create an institution",
+    status_code=status.HTTP_201_CREATED,
+)
+def create_institution(
+    institution: NewInstitutionDetails = Body(
+        ...,
+        title="Institution details",
+        description="Institution details for the institution to create.",
+    )
+) -> _NewInstitutionDetails:
+    with UnitOfWork() as unit_of_work:
+        institution_service = services.institution_service(unit_of_work.connection)
+        try:
+            institution_service.create(institution)
+
+            unit_of_work.commit()
+        except ValueError:
+            return institution_service.get_institution_by_name(
```

Comment:
If there is an error, I think the API should respond with an error, not with an existing institution. (Also, given that ValueError is a fairly generic exception type, I think it is perfectly reasonable to assume that it might have been caused by a reason other than the institution name existing already.)